### PR TITLE
Add docs into the client app permissions code

### PR DESCRIPTION
**Pull request by emaxx-google**
Better late than never.

### CODE REVIEW

File: third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/checker.js [hunk 1]
```diff
@@ -45,7 +45,18 @@ var PermissionsChecking =
     GSC.PcscLiteServerClientsManagement.PermissionsChecking;
 
 /**
- * FIXME(emaxx): Write docs.
+ * Provides interface for performing permissions checks for the given client
```
Comment (alex292) [inline third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/checker.js]: nit: Provides an interface for performing permisssion checks...

File: third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/checker.js [hunk 2]
```diff
@@ -66,8 +77,12 @@ Checker.prototype.logger = GSC.Logging.getScopedLogger(
     'PcscLiteServerClientsManagement.PermissionsChecking.Checker');
 
 /**
- * FIXME(emaxx): Write docs.
- * @param {string|null} clientAppId
+ * Starts the permission check for the given client app.
+ *
+ * The result is returned asynchronously as a promise (which will be eventually
```
Comment (alex292) [inline third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/checker.js]: nit: eventually be

File: third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/known-apps-registry.js [hunk 3]
```diff
@@ -69,7 +69,10 @@ PermissionsChecking.KnownApp = function(id, name) {
 var KnownApp = PermissionsChecking.KnownApp;
 
 /**
- * FIXME(emaxx): Write docs.
+ * This class provides interface for querying the information about the known
```
Comment (alex292) [inline third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/known-apps-registry.js]: nit: provides an interface

File: third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/known-apps-registry.js [hunk 4]
```diff
@@ -93,7 +96,11 @@ KnownAppsRegistry.prototype.logger = GSC.Logging.getScopedLogger(
     'PcscLiteServerClientsManagement.PermissionsChecking.KnownAppsRegistry');
 
 /**
- * FIXME(emaxx): Write docs.
+ * Starts the request of the information about the given app from the config
```
Comment (alex292) [inline third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/known-apps-registry.js]: nit: Requests information about...

File: third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/known-apps-registry.js [hunk 5]
```diff
@@ -118,8 +125,14 @@ KnownAppsRegistry.prototype.getById = function(id) {
 };
 
 /**
+ * Similar to getById(), but performs a batch request for the list of given IDs
+ * at once.
+ *
+ * The result is returned asynchronously as a promise that, when resolved, will
+ * contain the same number of elements as |idList|, with each value containing
```
Comment (alex292) [inline third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/known-apps-registry.js]: I assume the entries also have the same order in the returned list?

File: third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/managed-registry.js [hunk 6]
```diff
@@ -45,7 +45,8 @@ var PermissionsChecking =
     GSC.PcscLiteServerClientsManagement.PermissionsChecking;
 
 /**
- * FIXME(emaxx): Write docs.
+ * This class encapsulates the part of the client app permission checks that is
```
Comment (alex292) [inline third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/managed-registry.js]: check that is
or
checks that are

File: third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/managed-registry.js [hunk 7]
```diff
@@ -75,7 +76,11 @@ ManagedRegistry.prototype.logger = GSC.Logging.getScopedLogger(
     'PcscLiteServerClientsManagement.PermissionsChecking.ManagedRegistry');
 
 /**
- * FIXME(emaxx): Write docs.
+ * Starts the permission check for the given client app against the managed
+ * storage, that contains the admin-configured policy.
+ *
+ * The result is returned asynchronously as a promise (which will be eventually
```
Comment (alex292) [inline third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/managed-registry.js]: nit: eventually be

File: third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/user-prompting-checker.js [hunk 8]
```diff
@@ -65,7 +65,9 @@ var PermissionsChecking =
     GSC.PcscLiteServerClientsManagement.PermissionsChecking;
 
 /**
- * FIXME(emaxx): Write docs.
+ * This class encapsulates the part of the client app permission checks that is
```
Comment (alex292) [inline third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/user-prompting-checker.js]: same here
check that is
or
checks that are

File: third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/known-apps-registry.js (hunk 5 above)
Comment (emaxx-google) [reply third_party/pcsc-lite/naclport/server_clients_management/src/permissions_checking/known-apps-registry.js]: Good point. Yes, and I've reworded the comment to mention this.

**Pull request merged**